lib/server.go

```diff
@@ -2813,13 +2798,70 @@ func (srv *Server) _startAddressRelayer() {
 	}
 }
 
+func (srv *Server) _getFastHotStuffTransitionCheckInterval() <-chan time.Time {
+	// If the FastHotStuffConsensus does not exist, then there is nothing to do.
+	if srv.fastHotStuffConsensus == nil {
+		return nil
+	}
+
+	// If the FastHotStuffConsensus is already running, then there is nothing to do.
+	if srv.fastHotStuffConsensus.IsRunning() {
+		return nil
+	}
+
+	// If the FastHotStuffConsensus is exists but isn't running, then we need to wait
+	// and see if it's ready to initialize.
+	return time.After(1 * time.Minute)
+}
+
 func (srv *Server) _getFastHotStuffConsensusEventChannel() chan *consensus.FastHotStuffEvent {
 	if srv.fastHotStuffConsensus == nil {
 		return nil
 	}
 	return srv.fastHotStuffConsensus.fastHotStuffEventLoop.GetEvents()
 }
 
+func (srv *Server) tryTransitionToFastHotStuffConsensus() {
+	// If the FastHotStuffConsensus does not exist, then there is nothing to do.
+	if srv.fastHotStuffConsensus == nil {
+		return
+	}
+
+	// If the FastHotStuffConsensus is already running, then there is nothing to do.
+	if srv.fastHotStuffConsensus.IsRunning() {
+		return
+	}
+
+	// Exit early if the current tip height is below the final PoW block's height. We are ready to
+	// enable the FastHotStuffConsensus once we reach the final block of the PoW protocol. The
+	// FastHotStuffConsensus can only be enabled once it's at or past the final block height of
+	// the PoW protocol.
+	tipHeight := uint64(srv.blockchain.blockTip().Height)
```

Comment:
Do we need a chain lock to read this?

Reply:
Yes, it's good to hold the read lock for completeness